# Problem Set 9

## Part II  

In [1]:
import numpy as np
import matplotlib.pyplot as plt
import matplotlib as mpl
import sympy as sp

### Problem 1  

$$ \mathbf{F}(x,y) = \big(y^3-6y\big)\,\mathbf{i} + \big(6x - x^3\big)\,\mathbf{j}$$  

**a)**  Using Green's theorem, find the simple closed curve $C$ for which the integral $\displaystyle \oint_C\mathbf{F}\cdot\mathrm{d}\mathbf{r}$ (with positive oritentation) will have the largest positive value.  

$$ \mathrm{curl }\,\mathbf{F} = 6-3x^2 - 3y^2 + 6 = 12 - 3\big(x^2+y^2\big) $$

$$ \mathrm{curl }\,\mathbf{F} = 12 - 3r^2$$

Via Green's theorem,  

$$ \oint_C\mathbf{F}\cdot\mathrm{d}\mathbf{r} = \int\!\int_R 12\,\mathrm{d}A - 3\int\!\int_R r^2\,\mathrm{d}A$$

Suppose $R$ is bounded by a circle $C$ with a radius of $a$ centered at origin.  

$$ \int\!\int_R\mathrm{curl }\,\mathbf{F}\,\mathrm{d}A = 12\pi a^2 - 3\int\!\int_R r^2\,\mathrm{d}A$$

$$ 3\int\!\int_R r^2\,\mathrm{d}A = 3\int_0^{2\pi}\int_0^a r^3\,\mathrm{d}r\,\mathrm{d}\theta$$

$$ = \frac{3}{4}\int_0^{2\pi}a^4\,\mathrm{d}\theta = a^4\frac{3\pi}{2}$$

Thus,  

$$ \oint_C\mathbf{F}\cdot\mathrm{d}\mathbf{r} = 12\pi a^2 - a^4\frac{3\pi}{2}$$

Now, let  

$$ W:=W(a) = \oint_C\mathbf{F}\cdot\mathrm{d}\mathbf{r}$$

To find the max value of $W$, denoted as $W_\max$, we must find the roots of $W'(a)$.  

$$ W'(a) = 24\pi a - 6\pi a^3$$

$$ W'(a_e) = 0 \quad\implies\quad 4 - a_e^2 = 0  \quad\implies$$

$$ a_e^2 = 4 \quad\implies\quad a = 2$$

In [2]:
def dW(a): 
    return 24*np.pi*a - 6*np.pi*a**3

def W(a):
    return 12*np.pi*a**2 - (a**4)*3*np.pi/2

In [14]:
W(2)

75.39822368615503

Finally, $W=W_\max$ if and only if $C$ is a circle with a radius of $2$ centered at origin.  

$$ W_\max = W(2) = 48\pi - 24\pi = 24\pi$$

In [15]:
24*np.pi

75.39822368615503

**b)**  Compute this largest positive value.

As has been stated,  

$$ W_\max = W(2) = 24\pi \approxeq 75.4$$

### Problem 2

**a)**  Integrating throughout the continuity equation over an arbitrary region $R$ bounded by $C$, a positively oriented simple closed curve, gives:  

$$ \int\!\int_R \frac{\partial \rho}{\partial t}\,\mathrm{d}A + \int\!\int_R\mathrm{div }\,\mathbf{F}\,\mathrm{d}A = 0$$

And, via the above equation and Green's theorem in normal form,  

\begin{equation}\tag{2.2}
    \int\!\int_R \frac{\partial \rho}{\partial t}\,\mathrm{d}A + \oint_C \mathbf{F}\cdot\mathbf{n}\,\mathrm{d}s = 0
\end{equation}

It is important to note that the continuity equation seems to assume there is no external source of mass, which in this case means no external source of fluid (e.g. there is no water being poured onto a plane holding a shallow or -- ideally speaking -- 2D body of fluid.  Nor is there a sink carrying fluid out of the system.)  

In the case of a steady flow, this means that $\mathrm{div }\,\mathbf{F}= 0$, which means that, via Green's theorem in normal form, the rate of mass transport across any closed curve is equal to zero.  Such a condition can be realized a number of ways;  For example, a body of fluid at rest ($\mathbf{v} = \mathbf{0} \;\implies\; \mathbf{F}=\mathbf{0}$) or a body of fluid that rotates around origin satisfies $\mathrm{div }\,\mathbf{F} = 0$, where $\mathbf{F}$ is the flow field that applies to this body of fluid.

In the case of a non-steady flow, the continuity equation is still satisfied when $\displaystyle \frac{\partial \rho}{\partial t} = \mathrm{div }\,\mathbf{F}=0$, but there are other conditions that satisfy said equation;  Suppose that $\mathrm{div }\,\mathbf{F} \ne 0$.  Then the rate of mass transport (flux) across a closed curve $C$ is either positive (net flow away from the enclosed region $R$) or negative (net flow into $R$).  This can occur when the source (or sink) rate is not equal to zero (e.g. when fluid is being poured into the system from an external source), in which case the continuity equation does *not* apply.  It can also occur when the net change in density over time over or in $R$ is enough to explain the flux across $C$.  To understand this, one must understand the following:  

$$\frac{\partial \rho}{\partial t}(x,y,t)\,\mathrm{d}A $$

is equal to the instantaneous change in mass over time at $(x,y,t)$.  The double integral of the above over $R$ is equal to the instantaneous change in the total mass over or in $R$ over time at $t$.  Moreover, according to equation (2.2) and ultimately the continuity equation, this quantity must cancel out the rate of mass transport across $C$.  What this means is that a positive flux across $C$ (i.e. a net outward flow of fluid from $R$) must be accompanied by a decrease in the total mass over $R$, and vice versa, in order for the continuity equation to be satisfied.  Furthermore, if it is not satisfied, then, due to conversvation of mass, a source or sink adding or subtracting fluid to or from the system is at least partly responsible for the flux across $C$.

**b)**  $$ \mathrm{div}(g\mathbf{G}) = \frac{\partial g}{\partial x}M + g\frac{\partial M}{\partial x} + g_yN + gN_y $$

$$ = g\,\mathrm{div }\,\mathbf{G} + \nabla g\cdot\mathbf{G}$$

**c)**  Let $f$ and $\mathbf{r}$ be defined as:  

$$ f:=f(x,y,t) \quad\quad\text{and}\quad\quad \mathbf{r}:=\mathbf{r}(t) = \langle x(t),y(t)\rangle $$  

Recall that:  

$$ \frac{\mathrm{D}f}{\mathrm{D}t} = \frac{\mathrm{d}}{\mathrm{d}t}f\big(\mathbf{r}(t),t\big)$$

Also recall the chain rule.  Let $g:=g(s)$ and $s:=s(t)$.  Via the chain rule

$$ \frac{\mathrm{d}g(s(t))}{\mathrm{d}t} = \frac{\mathrm{d} g\circ s}{\mathrm{d}t} = \frac{\mathrm{d}g}{\mathrm{d}s}\frac{\mathrm{d}s}{\mathrm{d}t}$$

The total differential of $f$ is:  

$$ \mathrm{d}f = f_x\,\mathrm{d}x + f_y\,\mathrm{d}y + f_t\mathrm{d}t$$

Dividing throughout the above equation by $\mathrm{d}t$ gives:  

$$ \frac{\mathrm{d}f}{\mathrm{d}t} = f_x x'(t) + f_y y'(t) + f_t$$

And, when $f$ is contrained to the curve tracked by $\mathbf{r}$, as is the case in the definition of the convective derivative of $f$ with respect to time, we get:  

$$ \frac{\mathrm{D}f}{\mathrm{D}t} = \frac{\mathrm{d}}{\mathrm{d}t}f\big(\mathbf{r}(t),t\big) = \frac{\partial f}{\partial t}\big(\mathbf{r}(t),t\big) + \mathbf{v}\cdot\nabla f\big(\mathbf{r}(t),t\big)$$

where  

$$ \mathbf{r}'(t) = \langle x'(t),y'(t)\rangle := \mathbf{v}$$